x/escrow/time.go

```diff
@@ -0,0 +1,13 @@
+package escrow
+
+import "time"
+
+// currentTime is an alias that can be changed for the tests.
+var currentTime = func() time.Time {
```

Comment:
Nope. This is not allowed....

This must be deterministic and reproduceable. Imagine nodes had a clock drift of 5 seconds... it might be that some agree it was expired, others not, and they come to different results.

More so, imagine we sync this chain 2 months later... tx that were not expired when they ran the first time, would be expired now.

As mentioned in the issue, the *only* reference time you can use is from the block header.

You need to add this to the context (on the object, and also in BeginBlock of BaseApp), so it is available. Maybe you can make this `func isExpired(ctx weave.Context, t time.Time) bool` or even just extract the block header time from the context.

Reply:
True. I completely forgot about that.